include/util.h

```diff
@@ -135,10 +137,27 @@ int PURE strncmp(const char *s1, const char *s2, int n);
 long CONST char_to_long(char c);
 long PURE str_to_long(const char *str);
 
-// Library functions for counting leading/trailing zeros.
-// GCC's builtins will emit calls to these functions when the platform
-// does not provide suitable inline assembly.
-// We only emit function definitions if CONFIG_CLZL_IMPL etc are set.
+/* Library functions for counting leading/trailing zeros.
+ * GCC provides the following builtin function for this:
+ *   __builtin_clz(unsigned int x)
+ *   __builtin_clzl(unsigned long x)
```

Comment:
We don't currently use `clz` or `ctz` anywhere. We do use `clzll` for MCS on 32-bit platforms, and `ctzl` in some places. Perhaps just pick one example that we do use (the most important is `clzl`).
You've also buried the important bit about why these exist in the note about the GCC docs page. I'd prefer to keep that up front, especially since we know the GCC docs page is misleading.